### PR TITLE
Fix get input output variables

### PR DESCRIPTION
**Pull request by zaucker**
This adds two more routes and makes the current frontend working with the backend.
Webservivce testing is now done with the most current Agrammon model.

### CODE REVIEW

File: lib/Agrammon/Web/Routes.pm6 [hunk 1]
```diff
@@ -216,7 +216,7 @@ sub api-routes (Str $schema, $ws) {
         operation 'storeData', -> LoggedIn $user {
             request-body -> ( :datasetName($dataset-name), :$variable, :$value, :@branches, :@options , :$row) {
                 $ws.store-data(
-                    $user, $dataset-name, $variable, $value, @branches, @options, $row
+                    $user, $dataset-name, $variable, $value, @branches, @options, $row // Any
```
Comment (zaucker) [inline lib/Agrammon/Web/Routes.pm6:219]: @jnthn Is `// Any` the right thing to do here?

File: lib/Agrammon/Web/Service.pm6 [hunk 2]
```diff
@@ -112,8 +113,8 @@ class Agrammon::Web::Service {
             }
         }
 
-        use Agrammon::OutputFormatter::Text;
-        my $result = output-as-text($!model, $outputs, 'de', 'LivestockTotal');
+        # TODO: what is last parameter needed for?
+        my $result = output-as-text($!model, $outputs, 'de', 'LivestockTotal', True);
```
Comment (zaucker) [inline lib/Agrammon/Web/Service.pm6:117]: @jnthn This last parameter (`True`) is required, but seems to be unused in the called method (see chat).

File: share/agrammon.openapi [hunk 3]
```diff
@@ -369,6 +427,150 @@ components:
...
+                        - 'enum'
+                        - 'enum{ignore}'
+                units:
+                    description: Input labels
+                        schema:
+                            $ref: '#/components/schemas/LanguageLabels'
+                validator:
+                    description: Input validator
+                    type: object
+# TODO: can be either empty hash or must have name and args
```
Comment (zaucker) [inline share/agrammon.openapi:544]: @jnthn Do you know how this can be done in OpenAPI?

File: lib/Agrammon/Web/Service.pm6 (hunk 2 above)
Comment (jnthn) [reply lib/Agrammon/Web/Service.pm6:117]: Addressed in #188.

File: lib/Agrammon/OutputFormatter/GUI.pm6 [hunk 4]
```diff
@@ -25,7 +25,7 @@ sub get-data($model, $outputs) {
             for sorted-kv($_) -> $output, $raw-value {
                 my $format = $model.output-format($module, $output);
                 my $value = flat-value($raw-value);
-                my $formattedValue = $format ?? sprintf($format, $value)
+                my $formattedValue = ($format && $value) ?? sprintf($format, $value)
```
Comment (jnthn) [inline lib/Agrammon/OutputFormatter/GUI.pm6:28]: Do we never want to format `0`s? (If we do, maybe it should be `&& $value.defined`

File: lib/Agrammon/Web/Routes.pm6 (hunk 1 above)
Comment (jnthn) [reply lib/Agrammon/Web/Routes.pm6]: As discussed, better `Int :$row` or similar in the destrcuture.

File: lib/Agrammon/Web/Routes.pm6 [hunk 5]
```diff
@@ -251,12 +251,42 @@ sub api-routes (Str $schema, $ws) {
...
+                my %data = $ws.get-input-variables;
+                %data<datasetName> = $dataset-name;
+                # debug output on purpose
+                # dd "KEYS:", %data.keys;
+                # dd "GRAPHS:", %data<graphs>;
+                # dd "REPORTS:", %data<reports>;
+                # dd "INPUTS[0]:", %data<inputs>[0];
+                # for %data<inputs>[] -> $input {
+                #     dd $input<optionsLang>;
+                # }
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:276]: If it's going to stay around, consider putting it behind an AGRAMMON_DEBUG env var.

File: share/agrammon.openapi (hunk 3 above)
Comment (jnthn) [reply share/agrammon.openapi:544]: I think you're looking for [`oneOf` or `anyOf`](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/).

File: lib/Agrammon/Web/Routes.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6]: Done.

File: lib/Agrammon/Web/Service.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/Web/Service.pm6:117]: Will rebase shortly.

File: lib/Agrammon/OutputFormatter/GUI.pm6 (hunk 4 above)
Comment (zaucker) [reply lib/Agrammon/OutputFormatter/GUI.pm6:28]: Indeed, 0 is certainly a valid value. Fixed.

File: lib/Agrammon/Web/Routes.pm6 (hunk 5 above)
Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6:276]: I'll remember that suggestion, but removed that particular code as I hopefully won't need it anymore.

File: lib/Agrammon/Web/Service.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/Web/Service.pm6:117]: Rebase done.

File: share/agrammon.openapi (hunk 3 above)
Comment (zaucker) [reply share/agrammon.openapi]: The problem was, that I didn't know how to specify an empty hash/object, now I do:
```
EmptyHash:
    type: object
    properties: {}
```
Fixed.

File: share/agrammon.openapi [hunk 6]
```diff
@@ -1,642 +1,648 @@
...
-                                    type: string
-                                    enum:
-                                        - admin
-                                        - user
-            responses:
-                '200':
-                    description: Account created.
+    openapi: 3.0.0
+    info:
+        version: 1.0.0,
```
Comment (jnthn) [inline share/agrammon.openapi:3]: Why the extra level of indent here?

**Review by jnthn — APPROVED** [review]

**Pull request merged**